packages/go/dawgs/drivers/pg/driver.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2023 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package pg
+
+import (
+	"context"
+	"fmt"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"time"
+
+	"github.com/specterops/bloodhound/dawgs/graph"
+)
+
+var (
+	readOnlyTxOptions = pgx.TxOptions{
+		AccessMode: pgx.ReadOnly,
+	}
+
+	readWriteTxOptions = pgx.TxOptions{
+		AccessMode: pgx.ReadWrite,
+	}
+)
+
+type Config struct {
+	Options            pgx.TxOptions
+	QueryExecMode      pgx.QueryExecMode
+	QueryResultFormats pgx.QueryResultFormats
+}
+
+func OptionSetQueryExecMode(queryExecMode pgx.QueryExecMode) graph.TransactionOption {
+	return func(config *graph.TransactionConfig) {
+		if pgCfg, typeOK := config.DriverConfig.(*Config); typeOK {
+			pgCfg.QueryExecMode = queryExecMode
+		}
+	}
+}
+
+type Driver struct {
+	pool                      *pgxpool.Pool
+	schemaManager             *SchemaManager
+	defaultTransactionTimeout time.Duration
+}
+
+func (s *Driver) KindMapper() KindMapper {
+	return s.schemaManager
+}
+
+func (s *Driver) SetDefaultGraph(ctx context.Context, graphSchema graph.Graph) error {
+	return s.WriteTransaction(ctx, func(tx graph.Transaction) error {
+		return s.schemaManager.AssertDefaultGraph(tx, graphSchema)
+	})
+}
+
+func (s *Driver) SetBatchWriteSize(size int) {
+}
+
+func (s *Driver) SetWriteFlushSize(size int) {
+}
```

Comment:
Are these meant to be no-ops?

Reply:
No, I need to wire these up. I'll patch this in a moment.

Reply:
Wired up and note added for flush size